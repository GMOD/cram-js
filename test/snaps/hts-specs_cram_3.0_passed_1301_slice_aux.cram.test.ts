import { testFile } from '../testUtil'
testFile('hts-specs/cram/3.0/passed/1301_slice_aux.cram')
