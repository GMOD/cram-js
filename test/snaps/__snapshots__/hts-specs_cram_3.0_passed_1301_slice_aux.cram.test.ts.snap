// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole hts-specs/cram/3.0/passed/1301_slice_aux.cram without error 1`] = `
[
  {
    "fileId": "file:///nfs/users/nf",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 46,
      "_size": 20,
      "alignmentSpan": 0,
      "crc32": 591999209,
      "landmarks": [],
      "length": 1035,
      "numBases": 0,
      "numBlocks": 1,
      "numLandmarks": 0,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": -1,
    },
    "data": [
      {
        "_endPosition": 46,
        "_size": 20,
        "alignmentSpan": 0,
        "crc32": 591999209,
        "landmarks": [],
        "length": 1035,
        "numBases": 0,
        "numBlocks": 1,
        "numLandmarks": 0,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": -1,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "VN",
                "value": "1.6",
              },
              {
                "tag": "SO",
                "value": "unsorted",
              },
            ],
            "tag": "HD",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": false,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 5,
          },
          "parametersBytes": 1,
        },
        "BA": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 22,
          },
          "parametersBytes": 1,
        },
        "BF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 1,
          },
          "parametersBytes": 1,
        },
        "BS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 24,
          },
          "parametersBytes": 1,
        },
        "CF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 2,
          },
          "parametersBytes": 1,
        },
        "DL": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 26,
          },
          "parametersBytes": 1,
        },
        "FC": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 19,
          },
          "parametersBytes": 1,
        },
        "FN": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 17,
          },
          "parametersBytes": 1,
        },
        "FP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 18,
          },
          "parametersBytes": 1,
        },
        "HC": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 30,
          },
          "parametersBytes": 1,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 25,
            "stopByte": 9,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 9,
          },
          "parametersBytes": 1,
        },
        "MQ": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 16,
          },
          "parametersBytes": 1,
        },
        "NF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 8,
          },
          "parametersBytes": 1,
        },
        "NP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 11,
          },
          "parametersBytes": 1,
        },
        "NS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 10,
          },
          "parametersBytes": 1,
        },
        "PD": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 29,
          },
          "parametersBytes": 1,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 23,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 6,
          },
          "parametersBytes": 1,
        },
        "RI": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 3,
          },
          "parametersBytes": 1,
        },
        "RL": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 4,
          },
          "parametersBytes": 1,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 7,
            "stopByte": 9,
          },
          "parametersBytes": 2,
        },
        "RS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 27,
          },
          "parametersBytes": 1,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 28,
            "stopByte": 9,
          },
          "parametersBytes": 2,
        },
        "TC": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 14,
          },
          "parametersBytes": 1,
        },
        "TL": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 13,
          },
          "parametersBytes": 1,
        },
        "TN": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 15,
          },
          "parametersBytes": 1,
        },
        "TS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": true,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 1102,
      "_size": 21,
      "alignmentSpan": 300,
      "crc32": 3678966390,
      "landmarks": [
        181,
      ],
      "length": 1492,
      "numBases": 200,
      "numBlocks": 31,
      "numLandmarks": 1,
      "numRecords": 2,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1000,
    },
    "data": [
      {
        "_endPosition": 1102,
        "_size": 21,
        "alignmentSpan": 300,
        "crc32": 3678966390,
        "landmarks": [
          181,
        ],
        "length": 1492,
        "numBases": 200,
        "numBlocks": 31,
        "numLandmarks": 1,
        "numRecords": 2,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1000,
      },
      {
        "features": [
          {
            "alignmentStart": 1000,
            "cramFlags": 3,
            "flags": 99,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1200,
              "flags": 1,
              "readName": undefined,
              "sequenceId": 0,
            },
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 100,
            "readName": "match",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 300,
            "uniqueId": 1291,
          },
          {
            "alignmentStart": 1200,
            "cramFlags": 3,
            "flags": 147,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1000,
              "flags": 0,
              "readName": undefined,
              "sequenceId": 0,
            },
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 100,
            "readName": "match",
            "sequenceId": 0,
            "tags": {},
            "templateSize": -300,
            "uniqueId": 1292,
          },
        ],
        "header": {
          "_endPosition": 1576,
          "_size": 286,
          "compressedSize": 282,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            131,
            232,
            129,
            44,
            2,
            0,
            29,
            28,
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
            10,
            11,
            12,
            13,
            14,
            15,
            16,
            17,
            18,
            19,
            22,
            23,
            24,
            25,
            26,
            27,
            28,
            29,
            30,
            255,
            255,
            255,
            255,
            255,
            188,
            14,
            187,
            152,
            12,
            130,
            56,
            146,
            25,
            54,
            223,
            160,
            201,
            234,
            161,
            96,
            66,
            49,
            66,
            99,
            20,
            0,
            0,
            0,
            218,
            57,
            163,
            238,
            94,
            107,
            75,
            13,
            50,
            85,
            191,
            239,
            149,
            96,
            24,
            144,
            175,
            216,
            7,
            9,
            83,
            49,
            66,
            99,
            20,
            0,
            0,
            0,
            218,
            57,
            163,
            238,
            94,
            107,
            75,
            13,
            50,
            85,
            191,
            239,
            149,
            96,
            24,
            144,
            175,
            216,
            7,
            9,
            66,
            53,
            66,
            99,
            64,
            0,
            0,
            0,
            207,
            131,
            225,
            53,
            126,
            239,
            184,
            189,
            241,
            84,
            40,
            80,
            214,
            109,
            128,
            7,
            214,
            32,
            228,
            5,
            11,
            87,
            21,
            220,
            131,
            244,
            169,
            33,
            211,
            108,
            233,
            206,
            71,
            208,
            209,
            60,
            93,
            133,
            242,
            176,
            255,
            131,
            24,
            210,
            135,
            126,
            236,
            47,
            99,
            185,
            49,
            189,
            71,
            65,
            122,
            129,
            165,
            56,
            50,
            122,
            249,
            39,
            218,
            62,
            83,
            53,
            66,
            99,
            64,
            0,
            0,
            0,
            207,
            131,
            225,
            53,
            126,
            239,
            184,
            189,
            241,
            84,
            40,
            80,
            214,
            109,
            128,
            7,
            214,
            32,
            228,
            5,
            11,
            87,
            21,
            220,
            131,
            244,
            169,
            33,
            211,
            108,
            233,
            206,
            71,
            208,
            209,
            60,
            93,
            133,
            242,
            176,
            255,
            131,
            24,
            210,
            135,
            126,
            236,
            47,
            99,
            185,
            49,
            189,
            71,
            65,
            122,
            129,
            165,
            56,
            50,
            122,
            249,
            39,
            218,
            62,
            66,
            68,
            66,
            99,
            4,
            0,
            0,
            0,
            219,
            142,
            125,
            238,
            83,
            68,
            66,
            99,
            4,
            0,
            0,
            0,
            120,
            146,
            7,
            0,
          ],
          "contentId": 0,
          "contentPosition": 1290,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 1116907229,
          "parsedContent": {
            "_endPosition": 1160,
            "_size": 58,
            "contentIds": [
              1,
              2,
              3,
              4,
              5,
              6,
              7,
              8,
              9,
              10,
              11,
              12,
              13,
              14,
              15,
              16,
              17,
              18,
              19,
              22,
              23,
              24,
              25,
              26,
              27,
              28,
              29,
              30,
            ],
            "md5": [
              188,
              14,
              187,
              152,
              12,
              130,
              56,
              146,
              25,
              54,
              223,
              160,
              201,
              234,
              161,
              96,
            ],
            "numBlocks": 29,
            "numContentIds": 28,
            "numRecords": 2,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 300,
            "refSeqStart": 1000,
          },
          "uncompressedSize": 282,
        },
      },
    ],
  },
]
`;
